### ISSUE TITLE
Jackd managed by Rivendell can fail to start on boot. 

### ISSUE THREAD

**Issue by alexolivan** [ElvishArtisan/rivendell#784]
Hi.

I'm experimenting (Inspired by a mail on the mailing list) on having a rivendell 4 station to fully automatically recover from a reboot/reset/etc (the classic power outage scenario) an go to air.

One thing I'm finding is that, while rivendell.service very nicelly puts up and binds to a jackd instance in promiscuous mode, so everything is fantastic, it seems that it could fail to do so early on system boot (don't know why).
Restarting rivendell.service, after boot (or at any moment, once the system is booted) nicelly handles jackd daemon, as configured in RDAdmin/Host... but I'm looking for a way to ensure Rivendell's jackd managed daemon is started upon boot.

My early experiments was to put in the command a 'sleep 20; ' string, in order to chain it after a delay... but it doesn't work, and I guess the feature is designed to keep track of the pid of a single command put there... so I'm reluctant to point to a script there.

Any suggestions? Should I use a script instead of a command (I don't like the idea)? or may I be missing something?

Regards.

**Comment by fpeyraud** [comment]
Here, I start Jack as a SystemD service, and I override Rivendell systemD service so that it depends on jack service. It looks pretty stable here, but I agree there may be a race condition if JACK takes too much time to start and Rivendell tries to connect too early.

**Comment by ElvishArtisan** [comment]
Issues like this can be real bugbear troubleshoot and correct, not least because JACK is primarily aimed at scenarios where users explicitly start and stop it as part of a larger 'audio session'. We OTOH want to treat as a 'system service', wherein it runs without intermission so long as the host itself is booted up. Moving it into its own Systemd service group could solve a lot of these problems, but at the price of lots of additional complexity.

**Comment by alexolivan** [comment]
Hi Fred.

Probably nothing worth to worry... If strictly necessary, I agree that any skilled sys-admin would know how to revert/switch to have a dedicated jack unit service (In fact me too I did that before... I needed to have sys-wide promiscuous mode then).

Anyways, it would be nice to be able in RDAdmin to setup a time delay, to somehow give time to the system to end booting and settle... I'm but thinking on an 'average' end admin-user (but not a Linux sys-admin) that sticks to GUI management tools.

Probably the complication of coding and the risk of bugs doesn't pays off.
